[CL-2227] Rename the `content_builder` to `project_description_builder`

luucvanderzee opened the pull request:
Renamed everything specific to the project description builder from 'content builder' to 'project description builder'

I ran the E2E tests and all the ones related to the project description builder are still passing (the ones failing seem unrelated to these changes)

**Comments Adrien for the BE part:**

Currently, this feature flag is only used by the front-end to determine if the editor for project descriptions should be displayed. It has no impact on the back-end routes that allow for customizing project content or on the report builder functionality, which is controlled by a separate feature flag (`report_builder`).

It's possible that a `content_builder` feature flag would be useful in the future if we ever want to turn off all content builders at once, but that's not something we need at this time, and this feature flag was simply a misnomer.

luucvanderzee committed:
[CL-2227] Rename feature flag

luucvanderzee committed:
[CL-2227] Rename FE routes

luucvanderzee committed:
[CL-2227] Replace e2e ids

adessy committed:
[CL-2227] Rename the `content_builder` feature flag

Currently, this feature flag is only used by the front-end to determine if the editor for project descriptions should be displayed. It has no impact on the back-end routes that allow for customizing project content or on the report builder functionality, which is controlled by a separate feature flag (`report_builder`).

It's possible that a `content_builder` feature flag would be useful in the future if we ever want to turn off all content builders at once, but that's not something we need at this time, and this feature flag was simply a misnomer.

luucvanderzee committed:
Merge branch 'CL-2227-rename-content-builder' of github.com:CitizenLabDotCo/citizenlab into CL-2227-rename-content-builder

adessy committed:
[CL-2227] Make the file name for the feature flag clearer

luucvanderzee committed:
[CL-2227] Rename folders

luucvanderzee committed:
[CL-2227] _

luucvanderzee committed:
[CL-2227] Rename everything (WIP)

luucvanderzee committed:
[CL-2227] Rename e2e test files

luucvanderzee committed:
[CL-2227] _

luucvanderzee committed:
Merge branch 'CL-2227-rename-content-builder' of github.com:CitizenLabDotCo/citizenlab into CL-2227-rename-content-builder

luucvanderzee committed:
[CL-2227] Fix failing unit test

luucvanderzee committed:
Merge branch 'master' into CL-2227-rename-content-builder

adessy committed:
Merge branch 'master' into CL-2227-rename-content-builder

adessy committed:
[CL-2589] Fix seed data

`content_builder` feature flag has been renamed `project_description_builder`

adessy committed:
[CL-2589] Fix failing test

adessy committed:
Merge remote-tracking branch 'origin/CL-2227-rename-content-builder' into CL-2227-rename-content-builder

adessy committed:
[CL-2589] Remove dangling debug statement

luucvanderzee committed:
[CL-2227] Fix mock

sebastienhoorens reviewed (commented):
The changes look good, but did you test the migration? I think that with the current proposal, the enabled/allowed choices for existing tenants will be lost.

luucvanderzee commented:
@adessy ☝️

adessy commented:
Oops! I overlooked that small detail 😅🤦 I'll try to come up with a solution by tomorrow. But we'll probably have to roll this out in two steps.

amanda-anderson commented on front/app/modules/commercial/project_description_builder/admin/components/ContentViewer/Viewer/index.tsx:
```diff
@@ -0,0 +1,82 @@
+import React from 'react';
+
+// hooks
+import useProjectDescriptionBuilderLayout from '../../../../hooks/useProjectDescriptionBuilderLayout';
```
(optional) We can remove the " ./../../../ " here I think? :)  (I know this file was just moved though)

amanda-anderson commented on front/app/modules/commercial/project_description_builder/admin/components/ProjectDescriptionBuilderToggle/ProjectDescriptionBuilderToggle.test.tsx:
```diff
@@ -94,14 +101,14 @@ describe('ContentBuilderToggle', () => {
       { enabled: true }
     );
   });
   it('handles ContentBuilderLayout error', () => {
```
Missed renaming "ContentBuilderLayout" here I think? :)

luucvanderzee replied:
Yes, well spotted!

amanda-anderson commented on front/app/modules/commercial/project_description_builder/admin/components/ProjectDescriptionBuilderTopBar/ProjectDescriptionBuilderTopBar.test.tsx:
```diff
@@ -1,14 +1,14 @@
-const mockEditorData: IContentBuilderLayoutData = {
+const mockEditorData: IProjectDescriptionBuilderData = {
   id: '2',
   type: 'content_builder_layout',
```
(optional) Should we rename the type as well?

luucvanderzee replied:
I'm not sure... depends on how this is implemented in the backend. If this data structure is reusable for other content builders we can leave it like this, but if it's not it could also be renamed. WDYT @adessy ?

adessy replied:
Nope, it should stay as it is. All content builders will use the same type of layouts ✨

amanda-anderson reviewed (approved):
Looks great! Just a couple minor questions, but happy to approve in advance :)

Is there a way for me to do a quick functional review as well? Or is it not possible on this branch given a migration is needed?

luucvanderzee commented:
@amanda-anderson thanks for the review :) Will respond to your comments early next week. In the meanwhile you can do functional testing here: https://2227.epic.citizenlab.co/en/

amanda-anderson commented:
It's not related to this PR, but I've just noticed that the Desktop Preview is showing the mobile/tablet version, not the desktop :thinking: I wonder if this happened when we updated our breakpoints in the component library? Maybe we need to make the preview dimensions a bit larger so it's actually showing the Desktop view. It's definitely misleading for the admin :thinking: 

![image](https://user-images.githubusercontent.com/33987955/216629959-dd70c7c9-28cf-4972-a647-a6cf4f541a7a.png)

amanda-anderson commented:
Hmm also unrelated, as I see the same issue on Production.. but all of the multi-column layouts are missing the inner containers now (E.g. there used to be two dotted boxes _inside_ the two-column box, so it was clear you could drag content into either). Not sure when those disappeared :thinking: 
![image](https://user-images.githubusercontent.com/33987955/216630607-6682f93b-a964-432e-8347-29c4d9a5b285.png)

amanda-anderson commented:
Everything else I could test seems to be working well :+1: 

@luucvanderzee There were no existing content builder descriptions on the platform, so I wasn't able to test that scenario though.

luucvanderzee committed:
Merge branch 'master' into CL-2227-rename-content-builder

luucvanderzee committed:
[CL-2227] PR comments Amanda

luucvanderzee commented:
Thanks for the reviews @amanda-anderson ! I've solved two of the code issues with the third awaiting feedback by Adrien, and created two tickets in the Methods Jira board for the issues that you found up during functional testing :+1:

adessy committed:
[CL-2703] Add data migration to rename the content_builder flag

This might create some errors during the deployment/migration, but implementing a roll-out in multiple steps would take much more time.

adessy commented:
I wrote a quick data migration. @sebastienhoorens could you a look at it? It might produce some errors during the migration, but nothing too bad hopefully. And writing a multi-step roll-out would take much more effort.

adessy committed:
[CL-2703] Run the pending migration

adessy committed:
[CL-2703] Fix rubocop offenses

sebastienhoorens reviewed (approved):
Looks good :)

luucvanderzee committed:
Merge branch 'master' into CL-2227-rename-content-builder

luucvanderzee committed:
[CL-2227] Fix failing unit test

luucvanderzee committed:
Merge branch 'CL-2227-rename-content-builder' of
github.com:CitizenLabDotCo/citizenlab into CL-2227-rename-content-builder
